spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java

```diff
@@ -170,4 +174,31 @@ public static boolean useTimestampWithoutZoneInNewTables(RuntimeConfig sessionCo
     return false;
   }
 
+  /**
+   * Pulls any Catalog specific overrides for the Hadoop conf from the current SparkSession, which can be
+   * set via spark.sql.catalog.$catalogName.hadoop.*
+   *
+   * The SparkCatalog allows for hadoop configurations to be overridden per catalog, by setting
+   * them on the SQLConf, where the following will add the property "fs.default.name" with value
+   * "hdfs://hanksnamenode:8020" to the catalog's hadoop configuration.
+   *   SparkSession.builder()
+   *     .config(s"spark.sql.catalog.$catalogName.hadoop.fs.default.name", "hdfs://hanksnamenode:8020")
+   *     .getOrCreate()
+   * @param spark The current Spark session
+   * @param catalogName Name of the catalog to find overrides for.
+   * @return the Hadoop Configuration that should be used for this catalog, with catalog specific overrides applied.
+   */
+  public static Configuration hadoopConfCatalogOverrides(SparkSession spark, String catalogName) {
+    // Find keys for the catalog intended to be hadoop configurations
+    final String hadoopConfCatalogPrefix = String.format("%s.%s.%s", SPARK_CATALOG_CONF_PREFIX, catalogName, "hadoop.");
```

Comment:
I `"hadoop."` should be a `private static final String`.

Still not fan of using `hadoop.` instead of `override.` as mentioned on the other issue, since we will need to have the similar things for Hive as well for hive configurations too. This again will cause confusion for users who are using Hive and Spark as well.

Reply:
> I "hadoop." should be a private static final String.

Happy to update that.

> Still not fan of using hadoop. instead of override. as mentioned on the other issue, since we will need to have the similar things for Hive as well for hive configurations too. This again will cause confusion for users who are using Hive and Spark as well.

As for using `hadoop.` instead of `override.`, I do think that for spark users this is the most natural way to do it as it aligns with the way that spark offers to use `spark.hadop.*`, which many Spark users naturally look for.

As soon as we determine a better way to make it more generic across catalogs, I'm happy to champion either deprecation efforts of this, or to assist in any way I can with ensuring that both methods work for a reasonable period of time (whatever is decided).

Unfortunately, for now, I'm not sure if we manage to find any consensus around `override.` that would work with other catalogs. I'm happy to help look and try other methods in the source code.

As soon as we find something, I'm happy to help with implementing it, reviewing it, and ensuring the messaging gets out to the community as best as possible.

I would personally prefer to keep this version for Spark, as again this correlates really nicely with `spark.hadoop.*` which many Spark users naturally expect. So I would personally advocate for maintaining both methods, as spark users already have this method to override their job level hadoop configurations.

Reply:
Other than this I do not have any concerns wrt the configuration. I totally understand that in this case we can not find a good solution, so we just have to chose one.
I think after moving the prefix to a constant we can merge the PR. 

Thanks for the patience!
Peter

Reply:
I admittedly couldn't figure out what to name the `private static final String` for `hadoop.`, so I made a format string that is `private static final` as well as a `private static` function used to get the full prefix key for the catalog.

If you can think of a better name, by all means let me know. I thought maybe `SPARK_CATALOG_HADOOP_CONF_PREFIX_OVERRIDE_PART`, but that's terribly long.

Reply:
Please let me know if you think the current format string is better (or if you have any suggestions).

Having spent time looking through this, I'd be happy to try to find a more generic solution for the future, as generally speaking all engines likely have this request / need in some form or another.

Thanks again @pvary!